enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -485,7 +537,6 @@ async def restart_kernel(self, now=False, **kwargs):
             Any options specified here will overwrite those used to launch the
             kernel.
         """
-        self.restarting = True
```

Comment:
I think these (set and unset) need to remain in the `RemoteKernelManager` since things like auto-restart can trigger a restart that is not initiated by the user.